frontend/packages/client/src/api/notificationService.js

```diff
@@ -0,0 +1,117 @@
+import {
+  LP_APP_ID,
+  LP_CRO_KEY,
+  LP_EXPORT_KEY,
+} from 'api/constants';
+import Leanplum from 'leanplum-sdk';
+
+const COMMUNITY_UPDATES_CATEGORY_ID = 1;
+const options = {
+  method: 'GET',
+  headers: { accept: 'application/json' },
+};
+
+export const setUserId = async (walletId) => {
+  try {
+    Leanplum.setUserId(walletId);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const getUserSettings = async (walletId) => {
+  try {
+    fetch(
+      `https://api.leanplum.com/api?appId=${LP_APP_ID}&clientKey=${LP_EXPORT_KEY}&apiVersion=1.0.6&userId=${walletId}&action=exportUser`,
+      options
+    ).then((response) => {
+      let data = response.response[0];
+      let communitySubscriptions = [];
+      let isSubscribedToCommunityUpdates = true;
+      const res = {
+        email: data.userAttributes.email,
+      };
+
+      for (const property in data.userAttributes) {
+        if (property.includes('community')) {
+          let communityId = property.split('community')[1];
+          communitySubscriptions.push({
+            communityId,
+            subscribed: data.userAttributes[property] === 'True',
+          });
+        }
+      }
+
+      res.communitySubscription = communitySubscriptions;
+
+      if (data.unsubscribeCategories) {
+        data.unsubscribeCategories.forEach((category) => {
+          if (category.id == 1) {
+            isSubscribedToCommunityUpdates = false;
+          }
+        });
+      }
+
+      res.isSubscribedToCommunityUpdates = isSubscribedToCommunityUpdates;
+      return res;
+    });
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const setUserEmail = async (email) => {
+  try {
+    Leanplum.setUserAttributes({ email });
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const unsubscribeCommunity = async (communityId) => {
+  try {
+    let userAttributes = {};
+    let communityKey = 'community' + communityId;
+    userAttributes[communityKey] = false;
+    Leanplum.setUserAttributes(userAttributes);
+  } catch (e) {
+    throw new Error(e);
+  }
+};
+
+export const subscribeCommunity = async (communityId) => {
```

Comment:
on subscribe user is able to subscribe to All communities they are currently watching, ref: FE PR: https://github.com/DapperCollectives/CAST/pull/632/files#diff-c041fdf295a07a252c342ce2bf4c38cdac20e7e542b2cbe9ef1ac820ee1d9e68R22 and [figma design](https://www.figma.com/file/1GRfRrjNki3vR4T1LfLkHi/CAST%3A-Explorations?node-id=501%3A6698)

Might need to extend this method to support subscribe to all communities by fetching list of community ids that user is watching.

Reply:
This was not included in the design and i'm not convinced it is necessary, do you want to discuss to clarify what the need is here for me?

Reply:
as discussed, "subscribe to All" has been descoped, it's ok to leave it as it is for now.